src/Http/Controllers/PageController.php

```diff
@@ -10,8 +10,13 @@
 
 class PageController extends BaseController
 {
-    public function __invoke(Request $request, Page $page)
+    public function __invoke(Request $request, Page $page, Page $child_page, Page $grand_child_page)
```

Comment:
In addition to possibly allowing null parameters for child / grand child, you'll want to make sure the parent links are correct.  If grand child exists, its parent must be child page.  If not, 404.  If child exists, its parent must be page.  If not, 404.

Depending on requirements, you may also need to verify that the last non-null parameter has no children.  For example, if page is a parent, is page viewable on its own?  Or only page/child?  If requirements are only leaf node pages are viewable, then 404 for invalid paths.